.github/workflows/APIDigesterCheck.yml

```diff
@@ -0,0 +1,37 @@
+name: Swift API Digester Check
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # This will run the action every night at midnight
+  workflow_dispatch:  # Allows manual triggering
+
+jobs:
+  check-swift-api-digester:
+    runs-on: macos-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      with:
+        fetch-depth: 0
+
+
+    - name: Check API Dihester
+      run: |
+        TEMP_DIR=$(mktemp -d)
+        echo "Temporary directory created at $TEMP_DIR"
+        SDK_PATH=$(xcrun --sdk macosx --show-sdk-path)
+        echo "SDK Path: $SDK_PATH"
+        
+        # Run swift-api-digester
+        swift api-digester -sdk "$SDK_PATH" -diagnose-sdk --input-paths api-dump-test/A.json --input-paths api-dump-test/B.json >> "$TEMP_DIR/api-digester-output.txt" 2>&1
+          
+        # Display the output
+        cat "$TEMP_DIR/api-digester-output.txt"
+        
+        if diff "$TEMP_DIR/api-digester-output.txt" api-dump-test/expected-result.txt; then
+          echo "The output matches the expected result."
+        else
+          echo "The output does not match the expected result."
```

Comment:
if it fails can we send a slack message to the swift-devops channel?

Reply:
Successfully added!